src/common/checksum.c

```diff
@@ -612,18 +613,10 @@ calc_csum_recx_with_map(struct daos_csummer *obj, size_t *csum_nr,
 		}
 	}
 
-	if (consumed_bytes < recx->rx_nr * rec_len) {
-		/** Nothing mapped for recx or tail unmapped */
-		bytes_to_skip = (recx->rx_nr * rec_len) - consumed_bytes;
-		daos_sgl_processor(sgl, false, idx, bytes_to_skip,
-				   sgl_process_nop_cb, NULL);
```

Comment:
you mean this daos_sgl_processor() possibly cause INFO log ""Requested more bytes than what's available in sgl"?
I am wondering how can it happen, for fetch, the code obj_iod_sgl_valid() -> obj_req_size_valid() already checked sgl's buffer is enough for iod, why here can find the sgl is not enough for the recx?